hw3-js-classes/inheritance-es6.js

```diff
@@ -0,0 +1,83 @@
+class Car {
+    constructor(name, model, year, color, maxSpeed, fuelCapacity = 60, fuelConsumption = 10) {
+        this.name = name;
+        this.model = model;
+        this.year = year;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.fuelCapacity = fuelCapacity;
+        this.fuelConsumption = fuelConsumption;
+    }
+    getFullName() {
+        return `${this.name} ${this.model}`;
+    }
+
+    getAge() {
+        return ((new Date()).getFullYear() - this.year);
+    }
+
+    changeColor(newColor) {
+        if (newColor === this.color) {
+            alert(`Авто уже покрашено в ${newColor} цвет`);
+        } else {
+            this.color = newColor;
+            alert(`Цвет изменен на ${newColor}`);
+        }
+    }
+
+    calculateWay(kilometers, fuel) {
+        if (fuel < 10) {
+            alert(`Осталовь всего ${fuel} л. топлива. Нужна дозаправка`);
+        };
+        alert(`Минимальнео время в пути ${(kilometers / this.maxSpeed).toFixed(2)} ч.`);
+        if ((kilometers / 100 * this.fuelConsumption) <= fuel) {
+            alert(`Дозаправка не требуется`);
+        } else {
+            alert(`Для совершения поездки на ${kilometers} километров необходимо еще ${((kilometers / 100 * this.fuelConsumption - fuel) / this.fuelCapacity).toFixed(2)} бака бензина`);
+        }
+    }
+}
+
+let firstCar = new Car('Hyundai', 'Sonata YF', 2011, 'белый', 120);
+
+class Bmw extends Car {
+    constructor(model, year, color, maxSpeed, panoramicRoof, fuelCapacity, fuelConsumption) {
+        super('BMW', model, year, color, maxSpeed, fuelCapacity, fuelConsumption);
+        this.panoramicRoof = panoramicRoof;
+    }
+
+    getDescription() {
+        console.log(`Представляем Вам ${this.name} ${this.model}. Модель ${this.year} года. Цвет ${this.color}. На вопрос Есть ли в ней панорамная крыша, мы ответим ${this.panoramicRoof}. Спешите купить!`)
+    }
+}
+
+class Lexus extends Car {
+    constructor(model, year, color, maxSpeed, climateControl, fuelCapacity = 60, fuelConsumption = 10) {
```

Comment:
`fuelCapacity` & `fuelConsumption` already have their default values in basic `Car` class
```suggestion
    constructor(model, year, color, maxSpeed, climateControl, fuelCapacity, fuelConsumption) {
```